CoreFoundation/Base.subproj/CFFileUtilities.c

```diff
@@ -1056,9 +1056,13 @@ CF_PRIVATE void _CFIterateDirectory(CFStringRef directoryPath, Boolean appendSla
             }
 
             assert(!stuffToPrefix && "prefix not yet implemented");
-            CFStringRef filePath = CFRetain(fileName);
-
             Boolean isDirectory = file.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY;
+            CFMutableStringRef filePath = CFStringCreateMutableCopy(kCFAllocatorSystemDefault, nameLen + 1, fileName);
+            if (appendSlashForDirectories && isDirectory) {
+		UniChar slash = '\\';
+		CFStringAppendCharacters(filePath, &slash, 1);
```

Comment:
Hmm, would it be possible to use `CFStringAppend` with `CFPreferredSlashStr` instead?

Reply:
Sure! Though not directly since it's a macro.